src/librustdoc/core.rs

```diff
@@ -383,6 +384,7 @@ pub fn run_core(search_paths: SearchPaths,
                 fake_def_ids: RefCell::new(FxHashMap()),
                 all_fake_def_ids: RefCell::new(FxHashSet()),
                 generated_synthetics: RefCell::new(FxHashSet()),
+                all_traits: tcx.all_traits(LOCAL_CRATE).to_vec(),
```

Comment:
I wonder if we can filter this down to just the ones with blanket impls? That may require some overhead here but would also cut down on the traits we iterate over when gathering impls.

Reply:
I'm not sure to understand you. I do this call here because it otherwise would need to be done on every item (and also because @eddyb thinks it's better this way, which I agree).

Reply:
Right, i understand why you wanted to cache the traits. I'm asking if it's possible to store *less* traits in this, by filtering out traits that don't have a blanket/generic impl.